core/arch/arm/include/kernel/user_ta.h

```diff
@@ -87,20 +88,28 @@ struct user_ta_store_ops;
 #ifdef CFG_WITH_USER_TA
 TEE_Result tee_ta_init_user_ta_session(const TEE_UUID *uuid,
 			struct tee_ta_session *s);
-TEE_Result tee_ta_register_ta_store(struct user_ta_store_ops *ops);
 #else
 static inline TEE_Result tee_ta_init_user_ta_session(
 			const TEE_UUID *uuid __unused,
 			struct tee_ta_session *s __unused)
 {
 	return TEE_ERROR_ITEM_NOT_FOUND;
 }
-
-static inline TEE_Result tee_ta_register_ta_store(
-			struct user_ta_store_ops *ops __unused)
-{
-	return TEE_ERROR_NOT_SUPPORTED;
-}
 #endif
 
+/*
+ * Registers a TA storage.
+ *
+ * Note prio must be unique per storage in order to avoid dependency on
+ * registration order. This is enforced by a deliberate linker error in
+ * case of conflict.
+ *
+ * Also note that TA storage is sorted lexicographically instead of
+ * numerically.
+ */
+#define TEE_TA_REGISTER_TA_STORE(prio) \
+	int __tee_ta_store_##prio __unused; \
```

Comment:
The answer is a few lines above:
```
 * Note prio must be unique per storage in order to avoid dependency on
 * registration order. This is enforced by a deliberate linker error in
 * case of conflict.
```
Since the linker is supposed to garbage collect these during the final linking there should be no overhead due to this.